### PR TITLE
Initial shared configuration for Code Climate analysis

### PR DESCRIPTION
**Pull request by mkostick**
Code Climate's new analysis engine allows for a global configuration
shared by all repositories.  Here's a first attempt at that.  Most of
this is copied from the individual configurations of our repositories
currently using the new analysis engine, while the tslint config comes
from another one of our repos.

### CODE REVIEW

File: config/.codeclimate.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    enabled: true
+  bundler-audit:
+    enabled: true
+  duplication:
+    enabled: true
+    config:
+      languages:
+      - ruby
+      - javascript
+      - python
```
Comment (ancharon) [inline config/.codeclimate.yml]: Do we care about Python config?

File: config/.rubocop.yml [hunk 2]
```diff
@@ -0,0 +1,1169 @@
...
+                  number of characters. Use `delete` instead of `gsub` when
+                  you are deleting characters.
+  Reference: 'https://github.com/JuanitoFatas/fast-ruby#stringgsub-vs-stringtr-code'
+  Enabled: true
+
+##################### Rails ##################################
+
+Rails/ActionFilter:
+  Description: 'Enforces consistent use of action filter methods.'
+  Enabled: false
```
Comment (ancharon) [inline config/.rubocop.yml]: Looks like the remaining rubocop rules from here on down are disabled. Is this intentional?

File: config/.codeclimate.yml (hunk 1 above)
Comment (kwong-yw) [inline config/.codeclimate.yml]: We do have at least one project currently in Python, so it might be worthwhile to keep.

Comment (ancharon) [inline config/.codeclimate.yml]: Oh, I didn't see it when I took a quick look. Carry on, then. 👍

File: config/.rubocop.yml (hunk 2 above)
Comment (mkostick) [inline config/.rubocop.yml]: Yes, many of the available cops are ones we don't agree with.  I remember running rubocop against a few repos when setting this up initially (on the per-repo basis).

**Comment by mkostick** [comment]
I pushed an update to remove `.codeclimate.yml`.  Apparently *that* still has to be per-repo.

**Comment by mkostick** [comment]
Tweaked the `no-inferrable-types` setting.

**Comment by ancharon** [comment]
Did you want to add in the additional tslint rules @kwong-yw pointed out?

**Comment by kwong-yw** [comment]
I'm okay with adding the rest of the rules at a later point, when we have an actual project using them or when we have more :eyes: on whether or not we'd want them in use.

**Comment by ancharon** [comment]
Yeah, I think it would be worth going line-by-line through the JS style guide and pulling in the linting rules they recommend, too. That may be a bigger project than you want to tackle right now, though.

**Pull request merged**